src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2381,8 +2390,9 @@ private abstract class JsonLightResourceBaseScope : Scope, IODataJsonLightReader
                 IEdmStructuredType expectedResourceType,

```

Comment:
I think you should change "expectedResourceType" to accept "IEdmStructureTypeReference".

Reply:
Agree that this would be the preferred design; TypeReference encompasses Type and nullability; it would be nice if we could use that, rather than passing individually.

---
In reply to: [235490462](https://github.com/OData/odata.net/pull/1322#discussion_r235490462) [](ancestors = 235490462)

Reply:
Looks like you were able to make it work without too much churn?

---
In reply to: [237206849](https://github.com/OData/odata.net/pull/1322#discussion_r237206849) [](ancestors = 237206849)

Reply:
If you pass an IEdmStructuredTypeReference in to the function, you can still have an IEdmStructuredType property (IEdmStructuredTypeReference wraps the IEdmStructuredType).  But changing the signature to pass in an IEdmStructuredTypeReference would give you a more concise way of passing in the type and the nullability.

---
In reply to: [237206849](https://github.com/OData/odata.net/pull/1322#discussion_r237206849) [](ancestors = 237206849)